tests/network/dry_run/test_dry_run_kubemacpool.py

```diff
@@ -130,11 +117,12 @@ def dry_run_vm_with_mac_address(
 def test_dry_run_mac_not_saved(
     dry_run_vma_mac_address,
     vm_with_mac_address,
-    running_vm_mac_addresses,
 ):
     """Assure that the MAC address given to a VM created in dry-run mode is still available in the MAC-pool."""
-    assert dry_run_vma_mac_address == running_vm_mac_addresses, (
-        f"{vm_with_mac_address.name} MAC address is {running_vm_mac_addresses}"
+    vm_interfaces = vm_with_mac_address.get_interfaces()
+    secondary_net_mac = vm_interfaces[1][MAC_ADDRESS]
```

Comment:
Is there no need to check if `vm_interfaces` is not None / empty?

Reply:
It must be there because that was the spec with which the VM was created.
If it is not there, we wrote the test or/and there is a bug. Either way it will blow up in the same way access to the API will blow up if something is wrong.